replication-api-impl/src/main/java/com/connexta/replication/api/impl/data/AbstractPersistable.java

```diff
@@ -88,94 +110,146 @@ protected P writeTo(P pojo) {
+   */
+  protected <P> void convertAndSetOrFailIfNullOrEmpty(
+      String field,
+      Supplier<String> supplier,
+      Function<String, ? extends P> converter,
+      Consumer<? super P> consumer) {
+    consumer.accept(converter.apply(validateNotNullAndNotEmpty(field, supplier.get())));
+  }
+
   /**
    * Useful method that can be used to validate that the value of an enum field is defined as a
-   * valid string representation of an enumeration value before setting it in this object. If it is
-   * not a valid value than <code>unknown</code>> is used as the value to set in this object.
-   * Otherwise, the corresponding enum value is set accordingly. The pojo field's value is retrieved
-   * using a consumer (e.g. <code>pojo::getType</code>) and set in this object using a supplier
-   * (e.g. <code>this::setType</code>).
+   * valid string representation of an enumeration value before setting it in a given destination.
+   * If it is not a valid value than <code>unknown</code>> is used as the value to set in the
+   * destination. Otherwise, the corresponding enum value is set accordingly. The field's value is
+   * retrieved using a consumer (e.g. <code>pojo::getType</code>) and set in the destination using a
+   * supplier (e.g. <code>this::setType</code>).
    *
-   * @param field the name of the field being check for presence (used in the exception message)
+   * @param field the name of the field being checked for presence (used in the exception message)
    * @param clazz the field's enumeration class to convert to
-   * @param unknown the enumeration value to be used when the corresponding pojo field's string
-   *     value doesn't match any of the defined values
-   * @param supplier a supplier capable of retrieving the pojo's current value for the field
-   * @param consumer a consumer capable of updating this object with the field's value if it is
+   * @param unknown the enumeration value to be used when the corresponding field's string value
+   *     doesn't match any of the defined values
+   * @param supplier a supplier capable of retrieving the current value for the field
+   * @param consumer a consumer capable of updating the destination with the field's value if it is
    *     defined
    * @param <E> the field's enum type
-   * @throws ReplicationPersistenceException if the field's value as supplied by <code>supplier
+   * @throws InvalidFieldException if the field's value as supplied by <code>supplier
    *     </code> is <code>null</code> or empty
    */
   protected <E extends Enum<E>> void convertAndSetEnumValueOrFailIfNullOrEmpty(
       String field, Class<E> clazz, E unknown, Supplier<String> supplier, Consumer<E> consumer) {
     try {
       consumer.accept(Enum.valueOf(clazz, validateNotNullAndNotEmpty(field, supplier.get())));
       return;
-    } catch (IllegalArgumentException e) {
+    } catch (IllegalArgumentException e) { // ignore
```

Comment:
does naming it ignore make the finding go away?

Reply:
personal preference to keep `e` for exception variable and `t` for throwables inside catches. This came from the rebase since I had already fixed it on my branch.

Reply:
yes to @clockard 's question